src/tiledb/cloud/file_ingestion/ingestion.py

```diff
@@ -0,0 +1,226 @@
+    batch_size: Optional[int] = DEFAULT_BATCH_SIZE,
+    verbose: bool = False,
+) -> List[List[str]]:
+    """
+    Flatten and break a list of udf results into batches of a specified size.
+
+    :param udf_results: An iterable of iterables containing UDF results.
+    :param batch_size: Resulting chunk size, defaults to 100.
+    :return List[List[str]]: A list of chunks as lists.
+    """
+    logger = get_logger_wrapper(verbose)
+
+    # Flatten the results into a list.
+    flattened_results = [result for udf_result in udf_results for result in udf_result]
+
+    # Reduce batch size if there are fewer sample URIs
+    batch_size = min(batch_size, len(flattened_results))
+    num_chunks = ceil(len(flattened_results) / batch_size)
+    logger.info(f"Splitting results into {num_chunks} chunks...")
+
+    return [
+        flattened_results[n * batch_size : (n + 1) * batch_size]
+        for n in range(num_chunks)
+    ]
+
+
+def ingest_files_udf(
+    dataset_uri: str,
+    file_uris: Sequence[str],
+    destination: str,
+    *,
+    acn: Optional[str] = None,
+    config: Optional[dict] = None,
+    namespace: Optional[str] = None,
+    verbose: bool = False,
+) -> None:
+    """
+    Ingest files.
+
+    :param dataset_uri: The dataset URI.
+    :param file_uris: An iterable of file URIs.
+    :param destination: URI to ingest the files into.
+    :param acn: Access Credentials Name (ACN) registered in TileDB Cloud (ARN type),
+        defaults to None
+    :param config: Config dictionary, defaults to None.
+    :param namespace: TileDB-Cloud namespace, defaults to None.
+    :param verbose: Verbose logging, defaults to False.
+    """
+    logger = get_logger_wrapper(verbose)
+    # Trim potential trailing / in the destination URI
+    destination = destination.rstrip("/")
+
+    with tiledb.scope_ctx(config):
+        try:
+            if dataset_uri:
+                is_group = tiledb.object_type(dataset_uri) == "group"
+            else:
+                is_group = False
+        except Exception as exc:
+            # tiledb.object_type raises an exception
+            # if the namespace does not exist
+            logger.error(f"Error checking if {dataset_uri} is a Group. Bad namespace?")
+            raise exc
+
+        for file_uri in file_uris:
+            filename = sanitize_filename(os.path.basename(file_uri))
+            filestore_array_uri = f"{destination}/{filename}"
+            namespace = (
+                namespace or tiledb.cloud.user_profile().default_namespace_charged
+            )
+            logger.debug(
+                f"""
+                ---------------------------------------------
+                - Filename: {filename}
+                - Namespace: {namespace}
+                - Destination URI: {filestore_array_uri}
+                ---------------------------------------------"""
+            )
+
+            if not tiledb.array_exists(filestore_array_uri):
```

Comment:
I think we need to expose file overwrite options to the user or at least make sure that the behaviour is well documented. At the moment, this silently skips file overwrites without any documentation that this should be expected.

Reply:
@Shelnutt2 The current implementation produces a warning in the logs and moves on to the next file of the chunk (if any):  https://github.com/TileDB-Inc/TileDB-Cloud-Py/blob/55e4625a5435eee9ca2a4ddd5a0e8f27bd182714/src/tiledb/cloud/file_ingestion/ingestion.py#L122